src/scripts/main.js

```diff
@@ -1,3 +1,345 @@
+  <label>Position:
+    <input
+      name="position"
+      type="text"
+      data-qa="position"
+      required
+    >
+  </label>
+  <label>Office:
+    <select
+      name="office"
+      type="text"
+      data-qa="office"
+    >
+      <option value="Tokyo">Tokyo</option>
+      <option value="Singapore">Singapore</option>
+      <option value="London">London</option>
+      <option value="New York">New York</option>
+      <option value="Edinburgh">Edinburgh</option>
+      <option value="San Francisco">San Francisco</option>
+    </select>
+  </label>
+  <label>Age:
+    <input
+      name="age"
+      type="number"
+      data-qa="age"
+      required
+    >
+  </label>
+  <label>Salary:
+    <input
+      name="salary"
+      type="number"
+      data-qa="salary"
+      required
+    >
+  </label>
+  <button type="submit">Save to table</button>
+  `
+);
+
+body.append(form);
+
+let check = false;
+
+tableHeader.addEventListener('click', e => {
+  const target = e.target;
+
+  sortTable(target.cellIndex, target.innerHTML, check === target.cellIndex);
+
+  check = (check === target.cellIndex) ? false : target.cellIndex;
+});
+
+function sortTable(colNum, attribute, directSorting) {
+  const rowsArray = [...tbody.children];
+
+  let sorter;
+
+  switch (attribute) {
+    case 'Position':
+    case 'Name':
+    case 'Office':
+      sorter = function(rowA, rowB) {
+        const stringA = rowA.cells[colNum].innerHTML;
+        const stringB = rowB.cells[colNum].innerHTML;
+
+        return stringA.localeCompare(stringB);
+      };
+      break;
+
+    default:
+      sorter = function(rowA, rowB) {
+        const num1 = toNumber(rowA.cells[colNum].innerHTML);
+        const num2 = toNumber(rowB.cells[colNum].innerHTML);
+
+        return num1 - num2;
+      };
+  }
+
+  rowsArray.sort(sorter);
+
+  if (directSorting) {
+    tbody.append(...rowsArray.reverse());
+  }
+
+  tbody.append(...rowsArray);
+}
+
+const toNumber = function(string) {
+  const result = string.includes('$')
+    ? string.slice(1).split(',').join('')
+    : string;
+
+  return +result;
+};
+
+tbody.addEventListener('click', (e) => {
+  [...tbody.children].forEach(row => {
+    row.classList.remove('active');
+  });
+
+  e.target.parentElement.classList.add('active');
+});
+
+const pushNotification = (title, description, type) => {
+  const notification = document.createElement('div');
+
+  notification.classList.add('notification', type);
+  notification.dataset.qa = 'notification';
+
+  notification.innerHTML = `
+    <h2 class="title">${title}</h2>
+    <p>${description}</p>
+  `;
+
+  body.append(notification);
+
+  setTimeout(() => notification.remove(), 2000);
+};
+
+const errorText = (text) => {
+  pushNotification(
+    'Error!',
+    `${text}`,
+    'error'
+  );
+};
+
+const successText = (text) => {
+  pushNotification(
+    'Success!',
+    `${text}`,
+    'success'
+  );
+};
+
+const errorLong = 'It sould be at least 4 letters and must not include numbers';
```

Comment:
constants should be uppercased and placed at the top of the file
```suggestion
const ERROR_LONG = 'It sould be at least 4 letters and must not include numbers';
```